### PR TITLE
Adding Mintable, RBACMintable and Burnable ERC721 tokens

### PR DESCRIPTION
**Pull request by vittominacori**
# 🚀 Description

This add Mintable, RBACMintable and Burnable tokens to the ERC721 token lists.

Features:
- Added BurnableERC721Token
  - Only token owner can burn his own token
- Added MintableERC721Token
  - Only contract owner can mint new tokens 
  - Only contract owner can call the finishMinting function
- Added RBACMintableERC721Token
  - Only address with minter role can mint new tokens
  - Only owner can add/remove minter
  - All the MintableERC721Token stuffs

- [x] 📘 I've reviewed the [OpenZeppelin Contributor Guidelines](../blob/master/CONTRIBUTING.md)
- [x] ✅ I've added tests where applicable to test my new functionality.
- [x] 📖 I've made sure that my contracts are well-documented.
- [x] 🎨 I've run the JS/Solidity linters and fixed any issues (`npm run lint:all:fix`).

### CODE REVIEW

**Comment by shrugs** [comment]
@vittominacori  how do you think these two PRs should be combined?

https://github.com/OpenZeppelin/openzeppelin-solidity/pull/950

File: contracts/examples/SimpleERC721Token.sol [hunk 1]
```diff
@@ -0,0 +1,17 @@
+pragma solidity ^0.4.23;
+
+import "../token/ERC721/ERC721Token.sol";
+import "../token/ERC721/MintableERC721Token.sol";
+import "../token/ERC721/BurnableERC721Token.sol";
+
+
+// solium-disable-next-line max-len
+contract SimpleERC721Token is ERC721Token, MintableERC721Token, BurnableERC721Token {
+  constructor(string name, string symbol) public
```
Comment (shrugs) [inline contracts/examples/SimpleERC721Token.sol]: nitpick: formatting should look like

```
constructor(string _name, string _symbol)
  ERC721Token(_name, _symbol)
  public
  {}
```

**Comment by vittominacori** [comment]
@shrugs 

Regarding my code, I just wanted to replicate what I've already done with ERC20 RBACMintableToken.
So make the token able to be mint by more than an "owner".
And to do so I created `MintableERC721Token.sol`, `RBACMintableERC721Token.sol` and the behaviours to test them.

Your code is a little different, you allow the ERC721Minter to mint new tokens by validating signature.

About your code:

I think `RBACOwnable` should have only an owner and it is unuseful as the removed RBACWithAdmin contract. What's the difference between these 2 contracts and why one is being removed and the other should be created? In addition using the onlyOwner modifier to check Role can be confusing or maybe have conflict if people also extends from Ownable.

Also `RBACMintable` is more specific as requested. I mean why we should include all the possibile cases in the RBAC implementation and not only when requested? So I prefer to have the onlyMinter implementation in Token because of there are no other cases when the RBACMintable can be used so, for me, make no sense to have it into the RBAC folder.

PR could be merged but, guy, you are the reviewer so feel free to ask for changes in my code if it sounds repetitive or unuseful.

**Comment by shrugs** [comment]
The owner of a mintable token will be automatically set as the deployer because of the constructor in RBACOwnable, so I think it's ok for us not to automatically set the owner as the minter as well (although it would be good to mention that in the comments).

Let me know if you think it'd be easier to rebase this branch on #950 or #950 on this branch.

**Comment by vittominacori** [comment]
I think it could be better to merge #950 here. 
Keep MintableERC721 from my code, then rename and add the RBACMintableERC721 stuffs and then add your changes. 

The others files should not have conflicts.

Take a look to the added test behaviours and adapt to your token and minter contracts.

File: contracts/examples/SimpleERC721Token.sol (hunk 1 above)
Comment (canhlinh) [reply contracts/examples/SimpleERC721Token.sol]: I think so.

**Comment by canhlinh** [comment]
Awasome. I very much appreciate if  you guys continue working to finish this this PR.

**Comment by shrugs** [comment]
hey @vittominacori any update on your end? If you're unavailable, I can spend time this weekend and merge our PRs together. You'll still be the author of the original commits :)

**Comment by vittominacori** [comment]
Hi @shrugs I'm really busy these days. So it could be better if you can start merging PRs. 
Don't worry about authorship :)

Feel free to ask for changes, I will try to work on when I can.

**Comment by shrugs** [comment]
closing in favor of work on #1273, since this will need to use the new roles refactor branch.

**Pull request closed without merging**